inloop/solutions/templates/solutions/solution_detail.html

```diff
@@ -77,16 +130,101 @@ <h4 class="list-group-item-heading">{{ file.name }}</h4>
   </div><!-- /#files -->
 
   <div role="tabpanel" class="tab-pane" id="console">
-  {% if result.stdout or result.stderr %}
-    <pre class="console-output"><code>{{ result.stdout }}{{ result.stderr }}</code></pre>
-  {% else %}
+  {% if result.stdout %}
+    <pre class="console-output">
+      <code>{{ result.stdout }}</code>
+    </pre>
+  {% endif %}
+  {% if result.stderr %}
+    <pre class="console-output bg-danger text-white">
+      <code>{{ result.stderr }}</code>
+    </pre>
+  {% elif not result.stdout %}
     <p>Nothing to show here.</p>
   {% endif %}
   </div><!-- /#console -->
 
   <div role="tabpanel" class="tab-pane" id="unittests">
   {% include "solutions/includes/testsuites.html" %}
   </div><!-- /#unittests -->
+
+  <div role="tabpanel" class="tab-pane" id="checkstyle">
+
+  {% if checkstyle_files %}
+  {% for file in checkstyle_files %}
+
+  {% with file.checkstyle_warnings|length as warnings %}
+  {% with file.checkstyle_errors|length as errors %}
```

Comment:
Too much logic inside template -> should be a property or something similar.

Reply:
I fixed this in #c0dcc95